mixpanel.go

```diff
@@ -183,29 +219,38 @@ func (m *mixpanel) send(eventType string, params interface{}, autoGeolocate bool
 		return wrapErr(bodyErr)
 	}
 
-	if strBody := string(body); strBody != "1" && strBody != "1\n" {
-		return wrapErr(&ErrTrackFailed{Body: strBody, Resp: resp})
+	type verboseResponse struct {
+		Error  string `json:"error"`
+		Status int    `json:"status"`
+	}
+
+	var jsonBody verboseResponse
+	json.Unmarshal(body, &jsonBody)
+
+	if jsonBody.Status != 1 {
+		return wrapErr(&ErrTrackFailed{Message: jsonBody.Error})
 	}
 
 	return nil
 }
 
 // New returns the client instance. If apiURL is blank, the default will be used
 // ("https://api.mixpanel.com").
-func New(token, apiURL string) Mixpanel {
-	return NewFromClient(http.DefaultClient, token, apiURL)
+func New(token, secret, apiURL string) Mixpanel {
```

Comment:
See comment for `NewFromClient` and see if there's a way of making this non-breaking.